lib/facter/self_service.rb

```diff
@@ -130,4 +130,28 @@
     # Is Hiera 5 in use?
     { S0033: hiera_config_file.dig('version') == 5 }
   end
+  chunk(:S0039) do
+    # PuppetServer
+    next unless PuppetSelfService.primary?
```

Comment:
we want this confined to anywhere with a puppetserver running, so everything but  psql servers

Reply:
I have updated the readme and code as requested.

Reply:
you will want this running on primary, replica, compiler and legacy compilers